### ISSUE TITLE
When Entities are deleted, Relations still exist in the configuration.stc

### ISSUE THREAD

**Issue by wondie** [gltn/stdm#151]
The relation below still existed even though the entities Priority and Impact where removed. Below is the relation that was not removed. 
```
   <EntityRelation displayColumns="national_id" child="Priority" parent="Farmer" name="fk_ru_farmer_id_ru_priority_farmer" parentColumn="id" childColumn="farmer"/>
   <EntityRelation displayColumns="name,code" child="Priority" parent="check_input_service" name="fk_ru_check_input_service_id_ru_priority_input_service" parentColumn="id" childColumn="input_service"/>
   <EntityRelation displayColumns="name,code" child="Priority" parent="check_rank" name="fk_ru_check_rank_id_ru_priority_rank" parentColumn="id" childColumn="rank"/>

   <EntityRelation displayColumns="national_id" child="Impact" parent="Farmer" name="fk_ru_farmer_id_ru_impact_farmer" parentColumn="id" childColumn="farmer"/>
   <EntityRelation displayColumns="" child="socio_economic_impact" parent="Impact" name="fk_ru_impact_id_ru_socio_economic_impact_ru_impact_id" parentColumn="id" childColumn="ru_impact_id"/>

```

This resulted in the inability to use the Farmer entity that is referenced by the two tables.

**Comment by gkahiu** [comment]
Hi Wondie, I have tried to reproduce this issue by creating a child entity related to the party (in the STR) then deleted it but could not find any redundant relations in the configuration. Do you have a copy of the configuration you send it to me?

**Comment by wondie** [comment]
Hi John, 
The original stc. 
[configuration.zip](https://github.com/gltn/stdm/files/453239/configuration.zip)

The stc with deleted priority and impact entities. The relations still exists.
[configuration_2.zip](https://github.com/gltn/stdm/files/453241/configuration_2.zip)

**Comment by gkahiu** [comment]
Let me test, will revert shortly.

**Comment by gkahiu** [comment]
Should be fixed in [76f8648](https://github.com/gltn/stdm/commit/76f86488c5b28a1700422b22cb58f6f0c938667f) . The association entities were not being cleaned up when an entity was removed.

Please confirm if the issue is now fixed.

**Comment by wondie** [comment]
This issue is now fixed!
